create-klp-module: group .kpatch.symbols with like-scope

joe-lawrence opened the pull request:
From Oracle's Linker and Libraries Guide [1]:

"The symbols in a symbol table are written in the following order ...
The global symbols immediately follow the local symbols in the symbol
table. The first global symbol is identified by the symbol table sh_info
value. Local and global symbols are always kept separate in this manner,
and cannot be mixed together."

[1] https://docs.oracle.com/cd/E19120-01/open.solaris/819-0690/chapter6-79797/index.html

Fixes #854.
Signed-off-by: Joe Lawrence <joe.lawrence@redhat.com>

joe-lawrence commented:
Post-change symbol table:
[readelf --wide --symbols livepatch-test.ko](https://github.com/dynup/kpatch/files/2048896/readelf.txt)

jpoimboe commented on kpatch-build/create-klp-module.c:
```diff
@@ -93,6 +93,27 @@ static struct symbol *find_or_add_ksym_to_symbols(struct kpatch_elf *kelf,
+		struct symbol *s;
+
+		head = &kelf->symbols;
+		if (is_local_sym(sym)) {
```
Isn't this check redundant with line 101?

joe-lawrence commented:
v2:
- rebased on top of latest*
- removed redundant local symbol check
- creates the same symbol table as the one I attached for v1

\* I noticed github thinks there are now five changes here.  I didn't expect to see that as the other four changes are already merged.

joe-lawrence commented:
v3:
* rebase on top of the real master and not my out-of-date local master
